src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray_1.Builder.cs

```diff
@@ -239,6 +239,61 @@ public void Insert(int index, T item)
                 _elements[index] = item;
             }
 
+            /// <summary>
+            /// Inserts the specified values at the specified index.
+            /// </summary>
+            /// <param name="index">The index at which to insert the value.</param>
+            /// <param name="items">The elements to insert.</param>
+            public void InsertRange(int index, IEnumerable<T> items)
+            {
+                Requires.Range(index >= 0 && index <= this.Count, nameof(index));
+                Requires.NotNull(items, nameof(items));
+
+                int count = ImmutableExtensions.GetCount(ref items);
+                this.EnsureCapacity(this.Count + count);
+
+                if (index != this.Count)
+                {
+                    Array.Copy(_elements, index, _elements, index + count, _count - index);
+                }
+
+                if (!items.TryCopyTo(_elements, index))
+                {
+                    foreach (var item in items)
+                    {
+                        _elements[index++] = item;
+                    }
+                }
+
+                _count += count;
+            }
```

Comment:
Please remove all `this` if not explicitly required

Reply:
(everywhere in your code)

Reply:
Using `this` is preserving consistency with the rest of `System.Collections.Immutable` implementation. Even though it contradicts our style guide, it might be best to just preserve consistency (and possible remove all redundant `this` qualifications in a standalone PR).